packages/core/src/useForm/useFormActions.ts

```diff
@@ -170,32 +191,143 @@ export function useFormActions<TForm extends FormObject = FormObject, TOutput ex
     }
   }
 
-  async function reset(state?: Partial<ResetState<TForm>>, opts?: SetValueOptions) {
-    if (state?.values) {
-      form.setInitialValues(state.values, opts);
-    }
+  /**
+   * Typeguard for checking if a value is a Partial<ResetState<T>>.
+   *
+   * @template T - Type parameter for ResetState
+   * @param value - Value to check
+   * @returns True if value has ResetState properties
+   */
+  function isResetState<T>(value: unknown): value is Partial<ResetState<T>> {
+    return (
+      typeof value === 'object' &&
+      value !== null &&
+      !Array.isArray(value) &&
+      ('value' in value || 'touched' in value || 'revalidate' in value)
+    );
+  }
```

Comment:
I think this can be fragile if someone happens to have these object names randomly.

Long shot I know, but I will suggest another way to switch between the cases.